### PR TITLE
chore: update admin-endpoint-testing branch

### PR DESCRIPTION
**Pull request by BMarink512**
### 🛠 Changes being made
update the admin-endpoint-testing branch with the latest changes from main

### ✨ What's the context?
this branch implements some admin operations such as price updates, deletes, and their respective tests

### 🏎 Quality check

- [X] Are your changes following SOLID principles?

- [X] Did you make sure that there aren't any erroneous console logs, debuggers or leftover code in your changes?

- [X] Walk away, take a break, re-read what you filled out above does it make sense if you were coming in cold? What extra context could you provide?

### CODE REVIEW

File: src/book/book.controller.ts [hunk 1]
```diff
@@ -90,13 +94,14 @@ export class BookController {
   @ApiOperation({ summary: 'Delete book by id' })
   @ApiResponse({ status: 200, description: 'Book deleted', type: BookDto })
   @ApiResponse({ status: 404, description: 'Book not found' })
+  @UseGuards(AuthGuard, PermissionsGuard([AuthPermissions.UPDATE_ADMIN]))
   async deleteBook(@Param('id') id: number) {
     const deletedBook = await this.bookService.deleteBook(id)
     if (!deletedBook) {
       throw new NotFoundException(`Book with id ${id} not found`)
     }
     return {
```
Comment (AlfreMK) [inline src/book/book.controller.ts:103]: [REQ] We shouldn't delete the book, we should update it with isDeleted: true

File: src/book/book.controller.ts
Comment (AlfreMK) [inline src/book/book.controller.ts:1]: [REQ] We should hide (i.e remove the endpoint itself) or add a guard to the updateBook endpoint (probably we won't use it in the front anyway but just for safety)

Comment (BMarink512) [reply src/book/book.controller.ts:1]: ok, i'm going to add the guard

Along the same line, is there another feature the front will not implement?

Comment (BMarink512) [reply src/book/book.controller.ts:1]: also i'm going to add the guard to the populate endpoint

Comment (AlfreMK) [reply src/book/book.controller.ts:1]: We also will not implement all the features that have to do with assigning books as favorites.
With this admin endpoints and the rating we should be good with the backend (plus other details like fixing the prices from the scrapers)

**Review by AlfreMK — APPROVED** [review]
LGTM, Thank you for addressing the pending PR from iledantec

**Pull request merged**